service/history/workflowResetter.go

```diff
@@ -48,7 +48,14 @@ import (
 	"go.temporal.io/server/service/history/shard"
 )
 
+const (
+	ResetReapplyTypeAll ResetReapplyType = iota
+	ResetReapplyTypeNone
+)
+
 type (
+	ResetReapplyType int
```

Comment:
I removed all duplicated enums from code. Let's have design discussion before bringing it back. I am not convinced that it is better.

Reply:
>Let's have design discussion before bringing it back

i believe we already have some discussion, if you want we can have more